app/src/ui/lib/popover.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react'
+import FocusTrap from 'focus-trap-react'
+import { Options as FocusTrapOptions } from 'focus-trap'
+
+/**
+ * Position of the caret relative to the pop up. It's composed by 2 dimensions:
+ * - The first one is the edge on which the caret will rest.
+ * - The second one is the alignment of the caret within that edge.
+ *
+ * Example: TopRight means the caret will be in the top edge, on its right side.
+ **/
+export enum PopoverCaretPosition {
+  TopRight = 'top-right',
+  LeftTop = 'left-top',
+}
+interface IPopoverProps {
+  readonly onClickOutside: () => void
+  readonly caretPosition: PopoverCaretPosition
+}
+
+export class Popover extends React.Component<IPopoverProps> {
+  private focusTrapOptions: FocusTrapOptions
+  private divRef = React.createRef<HTMLDivElement>()
```

Comment:
Took me a minute to realize what this was for. Suggest a comment or maybe a name such as focusedDivRef?

Reply:
Yes, really good point!